drivers/sensor/htu21d/htu21d.c

```diff
@@ -0,0 +1,177 @@
+			return -EIO;
+		}
+		data->rh_sample &= 0xfffc;
+		/* RH% = -6 + 125 * sample / 2^16
+		 * Work in fixed-point 2 decimals
+		 */
+		const int sample = data->rh_sample & 0xfffc;
+		const int rh = ((12500 * sample) >> 16) - 600;
+
+		val->val1 = rh / 100;
+		val->val2 = (rh % 100) * 10000;
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int htu21d_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct htu21d_data *data = dev->data;
+	const struct htu21d_config *config = dev->config;
+	int res;
+	const uint8_t cmd_t[1] = { HTU21D_TRIGGER_T_NHM };
+	const uint8_t cmd_rh[1] = { HTU21D_TRIGGER_RH_NHM };
+	uint8_t buffer[3];
+	uint8_t crc;
+
+	if (chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	/* Reset samples so we don't read old values if this function fails */
+	data->t_sample = 0;
+	data->rh_sample = 0;
+
+	res = i2c_write(data->i2c,
+			cmd_t, sizeof(cmd_t),
+			config->i2c_addr);
+	if (res < 0) {
+		LOG_ERR("Failed to trigger sample: %d", res);
+		return -EIO;
+	}
+
+	/* Wait max 50 ms for 14-bit sample */
+	k_sleep(K_MSEC(50));
+	res = i2c_read(data->i2c,
+		       buffer, sizeof(buffer),
+		       config->i2c_addr);
+	crc = crc8(buffer, 2, 0x31, 0, false);
+	if (buffer[2] != crc) {
+		LOG_ERR("Checksum error");
+		return -EIO;
+	}
+	data->t_sample = (buffer[0] << 8) | buffer[1];
+
+	res = i2c_write(data->i2c,
+			cmd_rh, sizeof(cmd_rh),
+			config->i2c_addr);
+	if (res < 0) {
+		LOG_ERR("Failed to trigger sample: %d", res);
+		return -EIO;
+	}
+
+	/* Wait max 13 ms for 12-bit sample */
+	k_sleep(K_MSEC(13));
+	res = i2c_read(data->i2c,
+		       buffer, sizeof(buffer),
+		       config->i2c_addr);
+	crc = crc8(buffer, 2, 0x31, 0, false);
+	if (buffer[2] != crc) {
+		LOG_ERR("Checksum error");
+		return -EIO;
+	}
+	data->rh_sample = (buffer[0] << 8) | buffer[1];
+
+	return 0;
+}
+
+static const struct sensor_driver_api htu21d_driver_api = {
+	.sample_fetch = htu21d_sample_fetch,
+	.channel_get = htu21d_channel_get,
+};
+
+int htu21d_init(const struct device *dev)
+{
+	const struct htu21d_config *config = dev->config;
+	struct htu21d_data *data = dev->data;
+	int res;
+	const uint8_t reset_command[1] = { HTU21D_SOFT_RESET };
+
+	data->i2c = device_get_binding(config->i2c_bus);
+	if (!data->i2c) {
+		LOG_ERR("Could not get I2C device %s", config->i2c_bus);
+		return -EINVAL;
+	}
+
+	/* Soft-reset the sensor */
+	res = i2c_write(data->i2c,
+			reset_command, sizeof(reset_command),
+			config->i2c_addr);
+	if (res < 0) {
+		LOG_ERR("Failed to write I2C");
+		return -EIO;
+	}
+
+	/* Wait for reset */
+	k_sleep(K_MSEC(15));
+
+	return 0;
+}
+
+static struct htu21d_data htu21d_driver;
+static const struct htu21d_config htu21d_config = {
+	.i2c_bus = DT_INST_BUS_LABEL(0),
+	.i2c_addr = DT_INST_REG_ADDR(0),
+};
+
+DEVICE_DT_INST_DEFINE(0, htu21d_init, NULL,
+		      &htu21d_driver, &htu21d_config, POST_KERNEL,
+		      CONFIG_SENSOR_INIT_PRIORITY, &htu21d_driver_api);
```

Comment:
Sensors which are I2C/SPI bus devices must have multi-instance drivers (per @MaureenHelm; sensor driver maintainer); a single-instance driver like this is not allowed in new code.

Reply:
done